dio/lib/src/sentry_dio_extension.dart

```diff
@@ -12,8 +12,38 @@ extension SentryDioExtension on Dio {
   /// as well as request and response transformations.
   /// This must be the last initialization step of the [Dio] setup, otherwise
   /// your configuration of Dio might overwrite the Sentry configuration.
+  ///
+  /// You can also configure specific HTTP response codes to be considered
+  /// as a failed request. In the following example, the status codes 404 and 500
+  /// are considered a failed request.
+  ///
+  /// ```dart
+  /// dio.addSentry(
+  ///   failedRequestStatusCodes: [
+  ///     SentryStatusCode.range(400, 404),
+  ///     SentryStatusCode(500),
+  ///   ]
+  /// );
+  /// ```
+  ///
+  /// If empty request status codes are provided, all failure requests will be
+  /// captured. Per default, codes in the range 500-599 are recorded.
+  ///
+  /// If you provide failed request targets, the SDK will only capture HTTP
+  /// Client errors if the HTTP Request URL is a match for any of the provided
+  /// targets.
+  ///
+  /// ```dart
+  /// dio.addSentry(
+  ///   failedRequestTargets: ['my-api.com'],
+  /// );
+  /// ```
   void addSentry({
     Hub? hub,
+    List<SentryStatusCode> failedRequestStatusCodes = const [
+      SentryStatusCode.defaultRange()
+    ],
+    List<String> failedRequestTargets = const ['.*'],
```

Comment:
We can add a new const for `['.*']` so in case we have to replace it, we do it in a single place instead of 2

Reply:
Added a shares static const for both and also added failedRequestTargets to the `SentryHttpClient` ctor